scripts/clear_old_data.py

```diff
@@ -0,0 +1,179 @@
+    path: str
+    dt: datetime.datetime
+    file_type: FileType
+
+
+def create_smurf_date_dir(cfg: Config, path: str) -> Optional[FileInfo]:
+    dirname = os.path.split(path)[1]
+    try:
+        year = int(dirname[:4])
+        month = int(dirname[4:6])
+        day = int(dirname[6:])
+        dt = datetime.datetime(year=year, month=month, day=day)
+    except Exception:
+        if cfg.verbose:
+            print(f"Could not parse datetime: {dirname}")
+        return None
+    return FileInfo(path=path, dt=dt, file_type=FileType.SmurfData)
+
+
+def scan_smurf_data(cfg: Config) -> List[FileInfo]:
+    date_dirs: List[FileInfo] = []
+    base_dir = '/data/smurf_data'
+    now = datetime.datetime.now()
+    max_time_delta = datetime.timedelta(days=cfg.delete_smurf_data_after_days)
+    for d in os.listdir(base_dir):
+        result = create_smurf_date_dir(cfg, os.path.join(base_dir, d))
+        if result is not None:
+            if now - result.dt > max_time_delta:
+                date_dirs.append(result)
+
+    return sorted(date_dirs, key=lambda f:f.dt)
+
+
+def scan_timestream_dirs(cfg: Config) -> List[FileInfo]:
+    timestream_dirs: List[FileInfo] = []
+    base_dir = '/data/so/timestreams'
+    now = datetime.datetime.now()
+    max_time_delta = datetime.timedelta(days=cfg.delete_timestream_data_after_days)
+    for d in os.listdir(base_dir): # d is 5-digit ctime code
+        path = os.path.join(base_dir, d)
+        try:
+            # Give one day buffer for timezone effects, etc.
+            timestamp = (int(d) + 1)* 1e5
+            dt = datetime.datetime.fromtimestamp(timestamp)
+        except ValueError:
+            if cfg.verbose:
+                print(f"Could not parse datetime: {path}")
+            continue
+        file = FileInfo(
+            path=path,
+            dt=dt,
+            file_type=FileType.TimestreamData
+        )
+        if now - file.dt > max_time_delta:
+            timestream_dirs.append(file)
+    return sorted(timestream_dirs, key=lambda f:f.dt)
+
+def scan_core_dumps(cfg: Config) -> List[FileInfo]:
+    core_dump_dir = '/data/cores'
+    files: List[FileInfo] = []
+    now = datetime.datetime.now()
+    max_time_delta = datetime.timedelta(
+        days=cfg.delete_core_dumps_after_days
+    )
+    for f in os.listdir(core_dump_dir):
+        path = os.path.join(core_dump_dir, f)
+        ts = os.path.getctime(path)
+        dt = datetime.datetime.fromtimestamp(ts)
+        if now - dt > max_time_delta:
+            files.append(FileInfo(
+                path=path, dt=dt, file_type=FileType.CoreDump
+            ))
+    return files
+
+def scan_log_dirs(cfg: Config) -> List[FileInfo]:
+    log_dir = '/data/logs'
+    files: List[FileInfo] = []
+    now = datetime.datetime.now()
+    max_time_delta = datetime.timedelta(
+        days=cfg.delete_logs_after_days
+    )
+    for f in os.listdir(log_dir):
+        path = os.path.join(log_dir, f)
+        ts = os.path.getmtime(path)
```

Comment:
Why are you using `getmtime` here and `getctime` in the core dumps?

Reply:
There is not a huge difference, but I chose the mod-time here because the log-path is a directory, and I wanted the timestamp to be the time at which the last file was created or modified, instead of the creation time of the directory.

Reply:
Ahh gotcha, makes sense.